presto-orc/src/main/java/com/facebook/presto/orc/OrcZstdDecompressor.java

```diff
@@ -24,12 +25,19 @@
 {
     private final OrcDataSourceId orcDataSourceId;
     private final int maxBufferSize;
-    private final ZstdDecompressor decompressor = new ZstdDecompressor();
+    private final Decompressor decompressor;
 
-    public OrcZstdDecompressor(OrcDataSourceId orcDataSourceId, int maxBufferSize)
+    public OrcZstdDecompressor(OrcDataSourceId orcDataSourceId, int maxBufferSize, boolean zstdJniDecompressionEnabled)
     {
         this.orcDataSourceId = requireNonNull(orcDataSourceId, "orcDataSourceId is null");
         this.maxBufferSize = maxBufferSize;
+        if (zstdJniDecompressionEnabled) {
+            this.decompressor = (input, inputOffset, inputLength, output, outputOffset, maxOutputLength) -> (int) Zstd.decompressByteArray(output, 0, maxOutputLength, input, inputOffset, inputLength);
```

Comment:
nit
```java
this.decompressor = (input, inputOffset, inputLength, output, outputOffset, maxOutputLength) -> {
    long size = Zstd.decompressByteArray(output, 0, maxOutputLength, input, inputOffset, inputLength);
    if (Zstd.isError(size)) {
        throw new RuntimeException(Zstd.getErrorName(size));
    }
    return toIntExact(size);
};

Reply:
I am not quite sure if we should throw RuntimeException or MalformedInputException. 

Currently decompress catches the MalformInputException (which is a runtime exception) and rethrows it as OrcCorruptException which is a checked exception.

If we throw Runtime exception, then this behavior changes.